google-cloud-storage/src/test/java/com/google/cloud/storage/conformance/retry/ITRetryConformanceTest.java

```diff
@@ -0,0 +1,301 @@
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage.conformance.retry;
+
+import static com.google.cloud.storage.PackagePrivateMethodWorkarounds.blobCopyWithStorage;
+import static com.google.cloud.storage.PackagePrivateMethodWorkarounds.bucketCopyWithStorage;
+import static com.google.cloud.storage.conformance.retry.Ctx.ctx;
+import static com.google.cloud.storage.conformance.retry.State.empty;
+import static org.junit.Assert.assertNotNull;
+
+import com.google.cloud.conformance.storage.v1.InstructionList;
+import com.google.cloud.conformance.storage.v1.Method;
+import com.google.cloud.conformance.storage.v1.RetryTest;
+import com.google.cloud.conformance.storage.v1.RetryTests;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.conformance.retry.Functions.CtxFunction;
+import com.google.cloud.storage.conformance.retry.RetryTestFixture.CleanupStrategy;
+import com.google.cloud.storage.conformance.retry.RpcMethod.storage.buckets;
+import com.google.common.base.Charsets;
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.protobuf.util.JsonFormat;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.math.BigInteger;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+import java.util.function.BiPredicate;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import org.junit.AssumptionViolatedException;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class ITRetryConformanceTest {
+  private static final Logger LOGGER = Logger.getLogger(ITRetryConformanceTest.class.getName());
+
+  private static final String HEX_SHUFFLE_SEED_OVERRIDE =
+      System.getProperty("HEX_SHUFFLE_SEED_OVERRIDE");
+
+  private static final Set<RpcMethod> readBucketMethods =
+      ImmutableSet.of(buckets.get, buckets.list);
+  private static final List<List<String>> instructionSets =
+      ImmutableList.of(ImmutableList.of("return-503", "return-503") /*,
+      ImmutableList.of("return-400", "return-400")*/);
```

Comment:
Does this need to be in code and not fetched from the proto? Or is this a TODO?

Reply:
These are some pre-filters that are hanging around for a bit while we dig into the failing tests. Since the full tests suite is ~700 tests, we use these filters to pair down the set of tests that are even emitted to the test runner. Allowing for filtering like "Only run the cases where the instructions are [503, 503]".